composer/loggers/mosaic_logger.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Log to MPlatform."""
+
+from __future__ import annotations
+
+import logging
+import os
+import time
+from typing import Any, Dict, Optional
+
+from composer.loggers.logger import format_log_data_value
+from composer.loggers.logger_destination import LoggerDestination
+from composer.utils import MissingConditionalImportError, dist
+
+log = logging.getLogger(__name__)
+
+__all__ = ['MosaicLogger']
+
+
+class MosaicLogger(LoggerDestination):
+    """Log to MPlatform.
+
+    Logs metrics to MPlatform. Logging only happens on rank 0 every ``log_interval`` seconds to
```

Comment:
can logging only on rank 0 be optional? or is that an mcli limitation?

Reply:
MAPI limitation, otherwise we're going to start overwhelming it